specs/shared_inventory_openapi.yaml

```diff
@@ -21,48 +25,280 @@ paths:
-  /api/inventory/v1.0/resources/{resourceId}:
-    description: Base path for common inventory
+        "400":
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "401":
+          description: Unauthorized (really unauthenticated)
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "403":
+          description: Forbidden (really unauthorized)
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "500":
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+  /api/inventory/v1.0/resources/k8s-clusters:
+    description: Manage K8s Clusters
+    get:
+      summary: List K8s Clusters
+      operationId: listK8sClusters
+      # TODO: What parameters do we allow for querying?  Presumably Tags.  We need syntax and a convention or an
+      # algebra for tag expressions.
+      parameters:
+        - name: limit
+          in: query
+          description: How many items to return at one time (max 100)
+          required: false
+          schema:
+            type: integer
+            format: int32
+            minimum: 1
+            maximum: 100
+        - name: continue
+          in: query
+          description: The token for continuing to the next page of results
+          required: false
+          schema:
+            type: string
+      responses:
+        "200":
+          description: A paged list of kubernetes clusters registered in the current inventory.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/K8sClusterList'
+        "400":
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "401":
+          description: Unauthorized (really unauthenticated)
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "403":
+          description: Forbidden (really unauthorized)
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "500":
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+    post:
+      summary: Create a K8s Cluster
+      operationId: createK8sCluster
+      requestBody:
+        description: The K8s Cluster to create
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/K8sCluster"
+      responses:
+        "200":
```

Comment:
Should this return 200 or 202?
What is the thought process -

1. if we are going to accept the request and then process async, it better be 202. We really cannot generate a CI-cluster-id yet, can we?
2. Also if ACS and ACM (just making it up) tries to `POST` on same cluster at different times, we will not be creating 2 different resources, will we?